test/Microsoft.AspNet.Mvc.Core.Test/Filters/FormatFilterTest.cs

```diff
@@ -0,0 +1,378 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.PipelineCore;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.OptionsModel;
+using Microsoft.Net.Http.Headers;
+using Xunit;
+
+#if ASPNET50
+using Moq;
+using System.Net;
+#endif
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class FormatFilterTests
+    {
+        public enum FormatSource
+        {
+            RouteData,
+            QueryData,
+            RouteAndQueryData
+        }
+
+#if ASPNET50
+        [Theory]
+        [InlineData("json", FormatSource.RouteData, "application/json")]
+        [InlineData("json", FormatSource.QueryData, "application/json")]
+        [InlineData("json", FormatSource.RouteAndQueryData, "application/json")]
+        public void FormatFilter_ContextContainsFormat_DefaultFormat(
+            string format, 
+            FormatSource place, 
+            string contentType)
+        {
+            // Arrange  
+            var mediaType = MediaTypeHeaderValue.Parse("application/json");
+            var resultExecutingContext = CreateResultExecutingContext(
+                format, 
+                FormatSource.RouteData);            
+            var resourceExecutingContext = CreateResourceExecutingContext(
+                new IFilter[] { }, 
+                format, 
+                FormatSource.RouteData);
+            var filter = new FormatFilterAttribute();
+
+            // Act
+            filter.OnResourceExecuting(resourceExecutingContext);
+
+            // Assert
+            Assert.Null(resourceExecutingContext.Result);
+
+            // Act
+            filter.OnResultExecuting(resultExecutingContext);
+            
+            // Assert
+            var objectResult = Assert.IsType<ObjectResult>(resultExecutingContext.Result);
+            Assert.Equal(1, objectResult.ContentTypes.Count);
+            AssertMediaTypesEqual(mediaType, objectResult.ContentTypes[0]);
+        }
+
+        [Fact]
+        public void FormatFilter_ContextContainsFormat_InRouteAndQueryData()
```

Comment:
Add a brief comment here to make it clear to a reader what the result should be.

> if you set a format in both the route data and the query string, then the route data wins.